dbdimp.c

```diff
@@ -5917,78 +5939,77 @@ int mariadb_db_async_ready(SV* h)
 
 static bool parse_number(char *string, STRLEN len, char **end)
 {
-    int seen_neg = 0;
     bool seen_dec = FALSE;
     bool seen_e = FALSE;
-    bool seen_plus = FALSE;
-    char *cp;
-
-    if (len <= 0) {
-        len= strlen(string);
-    }
+    bool seen_digit = FALSE;
+    bool seen_second_neg = FALSE;
+    bool seen_second_plus = FALSE;
+    char *cp, *begin;
 
     cp= string;
 
     /* Skip leading whitespace */
-    while (*cp && isspace(*cp))
```

Comment:
Please explain, why you stop using ``is_space`` and ``is_digit``.

Reply:
In MySQL utf8mb4 charset are spaces defined as: ' ', '\t', '\n', '\v', '\f', '\r'. Fixed.